egs/librispeech/asr/simple_v1/mmi_att_transformer_train.py

```diff
@@ -96,9 +96,16 @@ def get_objf(batch: Dict,
             # scale less than one so it will be encouraged
             # to mimic ali_model's output
             ali_model_scale = 500.0 / (global_batch_idx_train // accum_grad + 500)
+            nnet_output_orig = nnet_output
             nnet_output = nnet_output.clone()  # or log-softmax backprop will fail.
             nnet_output[:, :,:min_len] += ali_model_scale * ali_model_output[:, :,:min_len]
 
+            x = nnet_output.abs().sum().item()
+            if x - x != 0:
+                print("Warning: reverting nnet output since it seems to be nan.")
+                nnet_output = nnet_output_orig
```

Comment:
@GNroy perhaps this is related to the error you had?  I found that sometimes I'd get NaN's in the forward pass of the alignment model.  I commented out `ali_model.eval()` as well as making this change, because I suspected that it had to do with test-mode batchnorm, but I might have been wrong, I need to test this.  It might relate to float16 usage (or a combination of the two).

Reply:
Thanks!
Actually, I resolved my issue.
NaNs were produced by the encoder part (not a loss or softmax problem as I thought before).
It was fixed with some hyperparameters re-tuning. In particular, setting eps=1e-3 for the optimizer helped.